composer/trainer/trainer.py

```diff
@@ -47,46 +50,74 @@
+            ``[constant_scheduler]``. (default: ``None``).
+
+            .. seealso:: :mod:`composer.optim.scheduler` for the different schedulers built into Composer.
+        device (str or Device, optional): The device to use for training. Either ``cpu`` or ``gpu``.
+            (default: ``cpu``)
         grad_accum (int, optional): The number of microbatches to split a per-device batch into. Gradients
             are summed over the microbatches per device. (default: ``1``)
-        grad_clip_norm (float, optional): The norm to clip gradient magnitudes to. Set to None for no gradient
+
+            .. note:: This is implemented by taking the batch yielded by the ``train_dataloader`` and splitting
+                it into ``grad_accum`` sections. Each section is of size ``train_dataloader // grad_accum``.
+                If the batch size of the dataloader is not divisible by ``grad_accum``,
+                then the last section will be of size ``batch_size % grad_accum``.
+        grad_clip_norm (float, optional): The norm to clip gradient magnitudes to. Set to ``None`` for no gradient
             clipping. (default: ``None``)
         validate_every_n_batches (int, optional): Compute metrics on evaluation data every N batches.
-             Set to -1 to never validate on a batchwise frequency. (default: ``-1``)
+             Set to ``-1`` to never validate on a batchwise frequency. (default: ``-1``)
         validate_every_n_epochs (int, optional): Compute metrics on evaluation data every N epochs.
-            Set to -1 to never validate on a epochwise frequency. (default: ``1``)
-        compute_training_metrics (bool, optional): True to compute metrics on training data and False to not.
+            Set to ``-1`` to never validate on a epochwise frequency. (default: ``1``)
+        compute_training_metrics (bool, optional): ``True`` to compute metrics on training data and ``False`` to not.
             (default: ``False``)
-        precision (str or Precision, optional): Numerical precision to use for training, one of 'fp32', 'fp16'
-            or 'amp' (recommended). (default: ``Precision.FP32``).
+        precision (str or Precision, optional): Numerical precision to use for training. One of ``fp32``, ``fp16``
+            or ``amp`` (recommended). (default: ``Precision.FP32``)
+
+            .. note::
+                ``fp16`` only works if ``deepspeed_config`` is also provided.
         scale_schedule_ratio (float, optional): Ratio by which to scale the training duration and learning rate
-            schedules. See :func:`scale_schedule` for details. (default: ``1.0``)
+            schedules. (default: ``1.0``)
+
+            .. seealso:: :func:`.scale_scheduler` for details on how scale scheduling
+                works.
         use_stepwise_schedulers (bool, optional): Whether schedulers will update after every optimizer step
```

Comment:
Can we rename this parameter to `step_schedulers_every_batch`? Stepwise is a bit unclear, since all schedulers have a step function.

"By default, stepwise schedulers are used when functional schedulers are provided, but not if only native Pytorch schedulers are provided (i.e. subclasses of torch.optim.lr_scheduler._LRScheduler). (default: ``None)" is formatting incorrectly. I would also rephrase this to say something like "By default, native pytorch schedulers (https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate) are updated every epoch, while :mod:`composer schedulers<composer.optim>` are updated every step. Just not a big fan of mentioning a private class, and we need to explain what a "functional scheduler" is,